src/promo/ThinBanner.jsx

```diff
@@ -0,0 +1,106 @@
+import Button from '../Button'
+import ArrowRightIcon from '@material-ui/icons/KeyboardArrowRight'
+import PropTypes from 'prop-types'
+import React from 'react'
+import withStyles from '@material-ui/core/styles/withStyles'
+import hexRgb from 'hex-rgb'
+
+const styles = theme => ({
+  button: {
+    backgroundColor: theme.palette.core && theme.palette.core.main,
+    borderRadius: 0,
+    color: [theme.palette.primary.contrastText],
+    fontWeight: 'bold',
+    justifyContent: 'left',
+    padding: '9px 12px', // special case, needs to match padding on existing topbar
+
+    '&:hover': {
+      backgroundColor: theme.palette.core && theme.palette.core.main
+    }
+  },
+  icon: {
+    position: 'absolute',
+    right: theme.spacing.unit
+  }
+})
+
+/**
+ * The `<ThinBanner>` component is a clickable block that can
+ * contain a short amount of text.
+ *
+ * Intended for use on mobile as an alternative to `<DonationBanner>`, it can
+ * also be used to promote surveys (traditionally with a green background) and
+ * newsletters (traditionally with a blue background).
+ *
+ * It optionally accepts both backgroundColor and color props. If provided, it'll
+ * use those. Without a backgroundColor, it'll default to TC red.
+ *
+ * Without a color, it'll determine the brightness of the background color, and if
+ * it's darker you'll see white text, if it's lighter you'll see dark grey text.
+ *
+ * ~~~js
+ * <ThinBanner>
+ *   Some nice text
+ * </ThinBanner>
+ * ~~~
+ *
+ */
+export const ThinBanner = ({ children, backgroundColor, color, classes, onClick }) => {
+  let textColor
+
+  if (color) {
+    textColor = `${color}`
+  } else if (backgroundColor) {
+    // Break the hex colour into component colours
+    const hexcolour = hexRgb(backgroundColor)
+    // Counting the perceptive luminance human eye favors green color...
+    const brightness = 1 - (0.299 * hexcolour.red + 0.5 * hexcolour.green + 0.114 * hexcolour.blue) / 255
```

Comment:
Would :heart: a comment with the source of this formula!

Found a StackOverflow comment about it https://stackoverflow.com/questions/596216/formula-to-determine-brightness-of-rgb-color#answer-596243 with a w3.org reference ("Techniques For Accessibility Evaluation And Repair Tools", https://www.w3.org/TR/AERT/#color-contrast).

Also, this could be a function :)

```js
/**
 * Perceived luminance for sufficient contrast
 * (https://www.w3.org/TR/AERT/#color-contrast)
 *
 * @param {String} color
 * @return {number}. Number between 0 and 1
 */
const perceivedLuminance = color => {
  const hexcolour = hexRgb(color)
  return 1 - (0.299 * hexcolour.red + 0.587 * hexcolour.green + 0.114 * hexcolour.blue) / 255
}
```

Reply:
Magic numbers! ✨

Reply:
Yes. Extracted to a function. Also left a comment with the URL to the stackoverflow where those magic numbers came from, so future-us can figure out what our intent was.